engine/table/src/main/java/io/deephaven/engine/table/impl/sources/UnionSourceManager.java

```diff
@@ -529,7 +531,10 @@ private void processExisting(@NotNull final Table constituent) {
                 nextSlotCurrFirstRowKey = nextSlotPrevFirstRowKey;
             }
 
-            final boolean needToProcessShifts = changes.shifted().nonempty() && constituent.getRowSet().isNonempty();
+            // Ignore shifts if the constituent was empty or became empty
+            final boolean needToProcessShifts = changes.shifted().nonempty()
+                    && constituent.getRowSet().isNonempty()
```

Comment:
On re-reading, maybe we should be using the same secondary conditions as when we apply a shift delta. That is, apply shifts if we have any and removed size isn't the same as previous size.